main/sceneManager.cpp

```diff
@@ -1,26 +1,63 @@
 #include "main/sceneManager.h"
+SceneManager::SceneManager(size_t collectors, double simulationRadius) : numCollectors_(collectors), simulationRadius_(simulationRadius)
+{
+    if (collectors % 4 != 0 || collectors - 1 % 4 != 0)
+    {
+        std::stringstream ss;
+        ss << "collector population:" << collectors << " % 4 or population - 1: " << collectors - 1 << " % 4 must be zero";
+        throw std::invalid_argument(ss.str().c_str());
+    }
+    // ( 2 *  pi * R ) / 2  for dividing whole half circle distance to equal distances
+    collectorRadius_ = constants::kPi * simulationRadius_ / numCollectors_;
+    createCollectors();
+}
+std::vector<objects::EnergyCollector *> SceneManager::getEnergyCollectors()
+{
+    std::vector<objects::EnergyCollector *> temp;
+    temp.reserve(energyCollectors_.size());
+    for (const auto &t : energyCollectors_)
+    {
+        temp.push_back(t.get());
+    }
+    return temp;
+}
 
-using Tri = objects::TriangleObj;
 
-void IamSimulator()
+// PRIVATE METHODS
+
+void SceneManager::createCollectors() // TODO: rewrite this, because it doesn't work. I messed uo with radians and degrees
 {
-    SceneManager manager;
-    CHECK(manager.loadModel("123231")) << "Cannot load model";
+    energyCollectors_.reserve(numCollectors_);
+    size_t numToGo;
 
-    std::vector<Tri> triangles = manager.getModelTriangles();
-    generators::PointSource source;
+    if (numCollectors_ % 2 == 0)
+    {
+        numToGo = numCollectors_;
+    }
+    else
+    {
+        energyCollectors_.push_back(std::make_unique<objects::EnergyCollector>(
+            core::Vec3(0, 0, simulationRadius_ / 2),
+            collectorRadius_));
+        numToGo = numCollectors_ - 1;
+    }
+    std::vector<core::Vec3> origins;
```

Comment:
1. Declare variables closest to where they are used.
2. Can't you just iterate on the temp. initializer list? (just checking, I do not know)